### PR TITLE
Fix logger

### PR DESCRIPTION
**Pull request by jahvon**
# Description

This PR bumps the go-utils version so that it included a fix in which we initialize the log config/logger when an environment variable is used to set the stats server log level. This was tested locally with a gloo image that pulled in these changes.

Also there are a few small docs updates.

# Context

1.7 introduced the ability to set a pod's log level via an env variable. However, we weren't initializing the logger in that case so logs were not being produced.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/content/reference/values.txt [hunk 1]
```diff
@@ -126,6 +126,7 @@
 |gloo.deployment.resources.requests.cpu|string||amount of CPUs|
 |gloo.serviceAccount.extraAnnotations.NAME|string||extra annotations to add to the service account|
 |gloo.serviceAccount.disableAutomount|bool||disable automunting the service account to the gateway proxy. not mounting the token hardens the proxy container, but may interfere with service mesh integrations|
+|gloo.logLevel|string||Level at which the pod should log. Options include "info", "debug", "warn", "error", "panic" and "fatal". Default level is info|
```
Comment (jahvon) [inline docs/content/reference/values.txt:129]: These are unrelated to the fix. I just realized that these docs didn't have this option.

**Review by saiskee — APPROVED** [review]
LGTM, thanks for adding log level to the values.go file!
